charms/tensorboard-controller/src/charm.py

```diff
@@ -1,184 +1,259 @@
-                                    "apiGroups": ["tensorboard.kubeflow.org"],
-                                    "resources": ["tensorboards/status"],
-                                    "verbs": ["get", "patch", "update"],
-                                },
-                                {
-                                    "apiGroups": ["tensorboard.kubeflow.org"],
-                                    "resources": ["tensorboards/finalizers"],
-                                    "verbs": ["update"],
-                                },
-                                {
-                                    "apiGroups": ["storage.k8s.io"],
-                                    "resources": ["storageclasses"],
-                                    "verbs": ["get", "list", "watch"],
-                                },
-                            ],
-                        }
-                    ]
-                },
-                "containers": [
-                    {
-                        "name": "deployment",
-                        "imageDetails": image_details,
-                        "command": ["/manager"],
-                        # "args": ["--enable-leader-election"],
-                        "ports": [{"name": "http", "containerPort": config["port"]}],
-                        "envConfig": {
-                            "ISTIO_GATEWAY": f"{gateway_ns}/{gateway_name}",
-                            "TENSORBOARD_IMAGE": "tensorflow/tensorflow:2.1.0",
-                        },
-                    }
-                ],
+        return ret_env_vars
+
+    @property
+    def _tensorboard_controller_layer(self) -> Layer:
+        """Create and return Pebble framework layer."""
+        layer_config = {
+            "summary": "tensorboard-controller layer",
+            "description": "Pebble config layer for tensorboard-controller",
+            "services": {
+                self._container_name: {
+                    "override": "replace",
+                    "summary": "Entrypoint of tensorboard-controller image",
+                    "command": self._exec_command,
+                    "startup": "enabled",
+                    "environment": self.service_environment,
+                    "on-check-failure": {PROBE_NAME: "restart"},
+                }
             },
-            {
-                "kubernetesResources": {
-                    "customResourceDefinitions": [
-                        {"name": crd["metadata"]["name"], "spec": crd["spec"]}
-                        for crd in yaml.safe_load_all(Path("files/crds.yaml").read_text())
-                    ],
-                },
+            "checks": {
+                PROBE_NAME: {
+                    "override": "replace",
+                    "period": "30s",
+                    "timeout": "20s",
+                    "threshold": 4,
+                    "http": {"url": f"http://localhost:{PROBE_PORT}{PROBE_PATH}"},
+                }
```

Comment:
I think we can ditch this check as we removed the code that actually uses it (`container.get_check()`)

Reply:
Resolving, see [here](https://github.com/canonical/kubeflow-tensorboards-operator/pull/69/files#r1230778793).